### PR TITLE
Adds new foreign route for vscode webview/webworker resources

### PR DESCRIPTION
**Pull request by jeanp413**
## Description

Adds new foreign route for vscode webview/weworker resources

## Related Issue(s)

Fixes #

## How to test

1. Open a workspace
2. Select vscode insiders
3. Check that webviews load sucessfully (open simple browser, go to extensions view and click on some extension so the extension editor shows the readme file of the extension)
4. Check web worker extension host is running sucessfully (install vim extension and it works)

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: components/ws-proxy/pkg/proxy/workspacerouter_test.go
Comment (akosyakov) [inline components/ws-proxy/pkg/proxy/workspacerouter_test.go:222]: i think this one should just go, port location should be already covered by other tests

Comment (akosyakov) [reply components/ws-proxy/pkg/proxy/workspacerouter_test.go:222]: ah, actually, it is test for webviews on port location for dev setup, how does it work now?

Comment (jeanp413) [reply components/ws-proxy/pkg/proxy/workspacerouter_test.go]: Added test for webviews on port location just in case it really shouldn't matter webworker runs on same origin in dev and webviews we could say it's useful but we can just force it to run from same origin too (we are changing webview code either way)

**Comment by jeanp413** [comment]
BTW think found some [unintended regex here](https://github.com/gitpod-io/gitpod/blob/4820e804bb86ecee9548ce9f1f9aca761d8436b9/install/installer/pkg/components/ws-proxy/configmap.go#L53) and it's [used here](https://github.com/gitpod-io/gitpod/blob/6345b948b4a5f9880bda95f7501e4e2dd562c2f3/components/ws-proxy/pkg/proxy/workspacerouter.go#L178)
The dot will match any character it should be escaped like the regex below it `WorkspaceHostSuffixRegex`

**Comment by akosyakov** [comment]
> Also I have [another branch](https://github.com/gitpod-io/gitpod/commit/2db7a76ba68bb89039b2eeca2646dfe4ce29264a) where I route the webviewurls to the same logic as blobserve, I think I remember you said last time that the cache could cause some issues, I checked codespaces and they return webview resources with cache enable too cache-control: public, max-age=31536000 cache will be invalidated as url changes per ide image tag/sha and uuid/hash subdomain, was there any other reason not to do it?

To improve caching we would like to move blobserve behind `ide.gitpod.io/blobserve` to decouple it from ws clusters.

**Review by geropl — APPROVED** [review]
Code LGTM, tested and works. :+1:

**Comment by jeanp413** [comment]
@gitpod-io/engineering-workspace is this a bug?:
> BTW think found some [unintended regex here](https://github.com/gitpod-io/gitpod/blob/4820e804bb86ecee9548ce9f1f9aca761d8436b9/install/installer/pkg/components/ws-proxy/configmap.go#L53) and it's [used here](https://github.com/gitpod-io/gitpod/blob/6345b948b4a5f9880bda95f7501e4e2dd562c2f3/components/ws-proxy/pkg/proxy/workspacerouter.go#L178)
The dot will match any character it should be escaped like the regex below it WorkspaceHostSuffixRegex

**Review by aledbf — APPROVED** [review]
LGTM

**Review by mustard-mh — APPROVED** [review]

**Comment by jeanp413** [comment]
Thanks, now why it doesn't get merged? :thinking:

**Comment by mustard-mh** [comment]
/hold
Because this PR is [blocking](https://github.com/gitpod-io/gitpod/pull/8709/commits/e7bee38dc4c3a8f2f32ce48633621f4ae36f4e57) merge [pool](https://prow.gitpod-dev.com/tide)

**Comment by jeanp413** [comment]
/unhold

@geropl @aledbf could you approve again please :pray: had to rebase to unblock it

**Review by geropl — APPROVED** [review]

**Review by aledbf — APPROVED** [review]

**Pull request merged**
